packages/spatial/src/renderer/functions/configureEffectComposer.ts

```diff
@@ -142,27 +161,96 @@ export const configureEffectComposer = (entity: Entity, schema?: typeof defaultP
       effects.push(eff)
+    } else if (key === Effects.ColorAverageEffect) {
+      const eff = new EffectClass(effectOptions.blendFunction)
+      composer[key] = eff
+      effects.push(eff)
+    } else if (key === Effects.OutlineEffect) {
+      const eff = new EffectClass(scene, camera, effectOptions)
+      composer[key] = eff
+      effects.push(eff)
+      //} else if (key == Effects.GodRaysEffect) {
+      //  const lightsource = null //tbd
+      //  const eff = new EffectClass(camera, lightsource, effectOptions)
+      //  composer[key] = eff
+      //  effects.push(eff)
+    } else if (key == Effects.LUT1DEffect) {
+      let lutPath = effectOptions.lut
+      if (lutPath == undefined) {
+        lutPath = null
+      }
+      let lut: Texture | null = null
+      if (lutPath != null) {
+        let textLoad = new TextureLoader()
+        //have to wait for the texture's image to load and then add the pass to the composer
+        lut = textLoad.load(lutPath, (texture) => {
+          const eff = new EffectClass(texture, effectOptions)
+          composer[key] = eff
+          effects.push(eff)
+          AddPass()
+        })
+      } else {
+        const eff = new EffectClass(lut, effectOptions)
+        composer[key] = eff
+        effects.push(eff)
+      }
+    } else if (key == Effects.LUT3DEffect) {
+      let lutPath = effectOptions.lut
+      if (lutPath == undefined) {
+        lutPath = null
+      }
+      let lut: Texture | null = null
+      if (lutPath != null) {
+        let textLoad = new TextureLoader()
+        //have to wait for the texture's image to load and then add the pass to the composer
+        lut = textLoad.load(lutPath, (texture) => {
+          const eff = new EffectClass(texture, effectOptions)
+          composer[key] = eff
+          effects.push(eff)
+          AddPass()
+        })
+      } else {
+        const eff = new EffectClass(lut, effectOptions)
+        composer[key] = eff
+        effects.push(eff)
+      }
+    } else if (key == Effects.PixelationEffect) {
+      const eff = new EffectClass(effectOptions.granularity)
+      composer[key] = eff
+      effects.push(eff)
+    } else if (key == Effects.ShockWaveEffect) {
+      const eff = new EffectClass(camera, effectOptions.position, effectOptions)
+      composer[key] = eff
+      effects.push(eff)
+    } else if (key == Effects.TextureEffect) {
+      let texturePath = effectOptions.texturePath
+      if (texturePath == undefined) {
+        texturePath = null
+      }
+      let texture: Texture | null = null
+      if (texturePath != null) {
+        let textLoad = new TextureLoader()
+        //have to wait for the texture's image to load and then add the pass to the composer
+        texture = textLoad.load(texturePath, (textureLoaded) => {
+          effectOptions.texture = textureLoaded
+          const eff = new EffectClass(effectOptions)
+          composer[key] = eff
+          effects.push(eff)
+          AddPass()
```

Comment:
I believe you're creating multiple duplicates of the EffectPass with `AddPass()`. We have to refactor this logic into PostProcessingComponent so we can use useEffects and the resource manager in order to load these textures.